pkg/clusterapi/aws/actuator.go

```diff
@@ -492,9 +492,12 @@ func (a *Actuator) updateStatus(machine *clusterv1.Machine, instance *ec2.Instan
 	return nil
 }
 
-func getClusterID(machine *clusterv1.Machine) (string, bool) {
-	clusterID, ok := machine.Labels[cov1.ClusterNameLabel]
-	return clusterID, ok
+func getClusterID(machine *clusterv1.Machine) (string, error) {
+	coMachineSetSpec, err := controller.MachineSetSpecFromClusterAPIMachineSpec(&machine.Spec)
+	if err != nil {
+		return "", err
+	}
+	return coMachineSetSpec.ClusterID, nil
```

Comment:
This appears broken as the cluster ID on the machine's provider config is an empty string. This is causing the get runinng instances to find nothing and infinitely create vms, until we hit the account limit.